### PR TITLE
[primTorch] special: ndtr, ndtri, log_ndtr, erfcx

### PR DESCRIPTION
**Pull request by khushi-411**
- Adds prims and _refs for `erfcx` and `ndtri`.
- Adds _refs for `ndtr`, and `log_ndtr`.

cc @kshitij12345 @lezcano @mruberry

### CODE REVIEW

File: torch/_refs/special/__init__.py [hunk 1]
```diff
@@ -45,6 +49,13 @@ def bessel_j1(a: TensorLikeType) -> TensorLikeType:
     return prims.bessel_j1(a)
 
 
+@_make_elementwise_unary_reference(
```
Comment (mruberry) [inline torch/_refs/special/__init__.py:52]: Why are the decorators for `erfcx` and `log_ndtr` below so different? `erfcx` supports `out`, too

File: torch/_refs/special/__init__.py [hunk 2]
```diff
@@ -93,6 +119,27 @@ def multigammaln(a: TensorLikeType, p: int) -> TensorLikeType:
...
 
 
+@register_decomposition(torch.ops.aten.special_ndtr)
+@out_wrapper()
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=utils.ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+)
+def ndtr(a: TensorLikeType) -> TensorLikeType:
+    a_sqrt_2 = a / 1.414213562373095048801688724209698
```
Comment (mruberry) [inline torch/_refs/special/__init__.py:129]: This doesn't look right -- shouldn't this expression be `a * M_SQRT1_2 `? 

https://github.com/pytorch/pytorch/blob/2a88f1b2d86a1fdc7380db768e67b18c24d199c4/aten/src/ATen/native/UnaryOps.cpp#L570

File: torch/_refs/special/__init__.py [hunk 3]
```diff
@@ -66,6 +77,21 @@ def i1e(a: TensorLikeType) -> TensorLikeType:
...
 
 
+@register_decomposition(torch.ops.aten.special_log_ndtr)
+@out_wrapper()
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=utils.ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+)
+def log_ndtr(a: TensorLikeType) -> TensorLikeType:
+    t = a * 0.707106781186547524400844362104849039
```
[the remark this answers is not in the document]
Comment (khushi-411) [reply torch/_refs/special/__init__.py]: Sure, Done!

File: torch/_refs/special/__init__.py (hunk 2 above)
Comment (khushi-411) [reply torch/_refs/special/__init__.py]: Hi, @mruberry! Thanks for the question. I believe the formula is right:

Here, it's `a * M_SQRT1_2`, where M_SQRT1_2 is `1/√2` = `0.70710678118654752440`.

And in my impl, I had written `a / √2.`

Comment (mruberry) [reply torch/_refs/special/__init__.py]: That makes sense but maybe the multiplication is preferable? Multiplication is typically faster than division

Comment (khushi-411) [reply torch/_refs/special/__init__.py]: I appreciate your input! I've already updated using multiplication :p Thank you :)

**Comment by khushi-411** [comment]
Hi, @mruberry! Gentle ping, please take another look. Thanks

**Comment by khushi-411** [comment]
Hi, Gentle ping, @mruberry!

**Review by mruberry — APPROVED** [review]
LGTM!

**Pull request closed without merging**